core/genesis.go

```diff
@@ -376,7 +376,11 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, g
 	if head == nil {
 		return newcfg, stored, errors.New("missing head header")
 	}
-	compatErr := storedcfg.CheckCompatible(newcfg, head.Number.Uint64(), head.Time)
+	genesisTimestamp := uint64(0)
+	if genesis != nil {
+		genesisTimestamp = genesis.Timestamp
+	}
+	compatErr := storedcfg.CheckCompatible(newcfg, head.Number.Uint64(), head.Time, genesisTimestamp)
```

Comment:
This is not ideal -- an alternative might be to have `storedcfg.CheckCompatible()` take a pointer to a genesisTimestamp, and then we can pass a nil pointer in in the case where the genesis itself is nil.